server/src/services/completion-service.ts

```diff
@@ -0,0 +1,349 @@
+            for (const [propName, propInfo] of Object.entries(classInfo.properties)) {
+                completions.push(
+                    this.createPropertyCompletion(className, propName, propInfo)
+                );
+            }
+        }
+    }
+
+    private addGlobalCompletions(
+        completions: CompletionItem[],
+        fileType: LanguageMode
+    ): void {
+        const functionsData = this.documentationService.getFunctions(fileType);
+        const classConstructors = this.documentationService.getClassConstructors(fileType);
+        const callbacksData = this.documentationService.getCallbacks(fileType);
+        const operatorsData = this.documentationService.getOperators();
+
+        // Add functions
+        for (const [funcName, funcInfo] of Object.entries(functionsData)) {
+            completions.push(this.createFunctionCompletion(funcName, funcInfo));
+        }
+
+        // Add class constructors
+        for (const [className, constructorInfo] of Object.entries(classConstructors)) {
+            completions.push(this.createConstructorCompletion(className, constructorInfo));
+        }
+
+        // Add callbacks (SLiM-only)
+        for (const [callbackName, callbackInfo] of Object.entries(callbacksData)) {
+            completions.push(this.createCallbackCompletion(callbackName, callbackInfo));
+        }
+
+        // Add operators
+        for (const [operatorName, operatorInfo] of Object.entries(operatorsData)) {
+            completions.push(this.createOperatorCompletion(operatorName, operatorInfo));
+        }
+    }
+
+    private createMethodCompletion(
+        className: string,
+        methodName: string,
+        methodInfo: MethodInfo
+    ): CompletionItem {
+        const cleanedSignature = cleanSignature(methodInfo.signature);
+        return {
+            label: methodName,
+            kind: CompletionItemKind.Method,
+            detail: cleanedSignature,
+            data: {
+                type: 'method',
+                className,
+                methodName,
+            },
+            command: {
+                title: 'Show Documentation',
+                command: 'slimTools.showFunctionDoc',
+                arguments: [`${className}.${methodName}`],
+            },
+        };
+    }
+
+    private createPropertyCompletion(
+        className: string,
+        propertyName: string,
+        propertyInfo: PropertyInfo
+    ): CompletionItem {
+        const cleanedType = cleanTypeNames(propertyInfo.type);
+        return {
+            label: propertyName,
+            kind: CompletionItemKind.Property,
+            detail: `Type: ${cleanedType}`,
+            data: {
+                type: 'property',
+                className,
+                propertyName,
+            },
+            command: {
+                title: 'Show Documentation',
+                command: 'slimTools.showPropertyDoc',
+                arguments: [`${className}.${propertyName}`],
+            },
```

Comment:
`showPropertyDoc` doesn't seem to be registered in `extension.ts`. It fails silently but I can see the issue in the console:

```
 ERR command 'slimTools.showPropertyDoc' not found: Error: command 'slimTools.showPropertyDoc' not found
	at Gy.h (file:///Applications/Visual%20Studio%20Code%202.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:114:42007)
```